Description: Upstream changeset 64324 and 64325 to fix build failure with gcc-7 (and 6.2.0-7 backported patch)

Index: virtualbox/src/VBox/Devices/USB/linux/USBProxyDevice-linux.cpp
===================================================================
--- virtualbox.orig/src/VBox/Devices/USB/linux/USBProxyDevice-linux.cpp
+++ virtualbox/src/VBox/Devices/USB/linux/USBProxyDevice-linux.cpp
@@ -110,8 +110,16 @@
  */
 typedef struct USBPROXYURBLNX
 {
-    /** The kernel URB data */
+    /** The kernel URB data. */
+#if RT_GNUC_PREREQ(6, 0)
+    /* gcc 6.2 complains about the [] member of KUrb */
+# pragma GCC diagnostic push
+# pragma GCC diagnostic ignored "-Wpedantic"
+#endif
     struct usbdevfs_urb             KUrb;
+#if RT_GNUC_PREREQ(6, 0)
+# pragma GCC diagnostic pop
+#endif
     /** Space filler for the isochronous packets. */
     struct usbdevfs_iso_packet_desc aIsocPktsDonUseTheseUseTheOnesInKUrb[8];
     /** Node to link the URB in of the existing lists. */
@@ -121,11 +129,11 @@
     /** The next linux URB if split up. */
     struct USBPROXYURBLNX           *pSplitNext;
     /** Don't report these back. */
-    bool                             fCanceledBySubmit;
+    bool                            fCanceledBySubmit;
     /** This split element is reaped. */
-    bool                             fSplitElementReaped;
+    bool                            fSplitElementReaped;
     /** Size to transfer in remaining fragments of a split URB */
-    uint32_t                         cbSplitRemaining;
+    uint32_t                        cbSplitRemaining;
 } USBPROXYURBLNX, *PUSBPROXYURBLNX;
 
 /**
Index: virtualbox/src/VBox/Main/src-server/linux/HostDnsServiceLinux.cpp
===================================================================
--- virtualbox.orig/src/VBox/Main/src-server/linux/HostDnsServiceLinux.cpp
+++ virtualbox/src/VBox/Main/src-server/linux/HostDnsServiceLinux.cpp
@@ -34,6 +34,11 @@
 
 #include <linux/limits.h>
 
+/* Workaround for <sys/cdef.h> defining __flexarr to [] which beats us in
+ * struct inotify_event (char name __flexarr). */
+#include <sys/cdefs.h>
+#undef __flexarr
+#define __flexarr [0]
 #include <sys/inotify.h>
 #include <sys/types.h>
 #include <sys/socket.h>
